### PR TITLE
gleam: 0.6.0 -> 0.7.1

### PR DESCRIPTION
**Pull request by NobbZ**
##### Motivation for this change

Upstream release of a new version

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/gleam/default.nix [hunk 1]
```diff
@@ -1,19 +1,19 @@
-{ stdenv, rustPlatform, fetchFromGitHub, Security }:
+{pkgs,  stdenv, rustPlatform, fetchFromGitHub, Security }:
```
Comment (filalex77) [inline pkgs/development/compilers/gleam/default.nix]: I don't see any usage of `pkgs` in this file. In nixpkgs, each derivation is called (in `pkgs/top-level/all-packages.nix`) as

```nix
{
  name = callPackage /path/to/derivation { };
}
```

where `callPackage` allows to select dependencies via derivation arguments. Thus, you can skip `pkgs`. If there is any reason for why it is added here, please let me know!

**Review by Br1ght0ne — COMMENTED** [review]
Just wanted to do this myself. Besides a nitpick, LGTM. Thanks!

File: pkgs/development/compilers/gleam/default.nix (hunk 1 above)
Comment (NobbZ) [reply pkgs/development/compilers/gleam/default.nix]: Of, course, I just inserted it to have access to `pkgs.lib.fakeSha256` during updating the hashes, it was not meant to slip through…

**Review by Br1ght0ne — APPROVED** [review]

**Comment by NobbZ** [comment]
In the meantime `0.7.1` has been released, though it reports itself as `0.7.0` as the tag is on the wrong commit.

This has been reported upstream and might cause either a re-release from the correct tag, or just stick where we are, depending on how large the commit gap actually is.

Depending on the merge status of this PR, I will just push a change after the upstream issue has been resolved or create a new PR.

**Comment by NobbZ** [comment]
Okay, 0.7.1 got re-tagged under another commit. I have adjusted the files accordingly.

**Comment by danieldk** [comment]
Closing this, since we are already at 0.8.0.

https://github.com/NixOS/nixpkgs/pull/86304

**Pull request closed without merging**